OnlineShop.Services.Basket.BusinessLayer/Services/Implementations/BasketService.cs

```diff
@@ -0,0 +1,101 @@
+using OnlineShop.Services.Basket.DataLayer.Repositories.Interfaces;
+
+namespace OnlineShop.Services.Basket.BusinessLayer.Services.Implementations
+{
+    public class BasketService : IBasketService
+    {
+        private readonly IBasketRepository _basketRepository;
+        private readonly IMapper _mapper;
+
+        public BasketService(IBasketRepository basketRepository, IMapper mapper)
+        {
+            _basketRepository = basketRepository;
+            _mapper = mapper;
+        }
+
+        public async Task<ResponseDto<BasketDto>> GetBasketAsync(string userId, CancellationToken cancellationToken = default)
+        {
+            var basket = await _basketRepository.GetBasketAsync(userId, cancellationToken);
+
+            if (basket is null)
+            {
+                throw new BasketNotFoundException(userId);
+            }
+
+            var basketDto = _mapper.Map<BasketDto>(basket);
+            basketDto.UserId = userId;
+
+            return new ResponseDto<BasketDto> { Result = basketDto };
+        }
+
+        public async Task<ResponseDto<BasketDto>> UpdateBasketAsync(string userId, UpdateBasketDto basketDto, CancellationToken cancellationToken = default)
+        {
+            
+            var updatedBasket = _mapper.Map<ShoppingCart>(basketDto);
+
+            updatedBasket = await _basketRepository.UpdateBasketAsync(userId, updatedBasket, cancellationToken);
+
+            var result = _mapper.Map<BasketDto>(updatedBasket);
+            result.UserId = userId;
+
+            return new ResponseDto<BasketDto> { Result = result, Message = "Basket updated successfully" };
+        }
+
+        public async Task<ResponseDto<object>> DeleteBasketAsync(string userId, CancellationToken cancellationToken = default)
+        {
+            await _basketRepository.DeleteBasketAsync(userId, cancellationToken);
+
+            return new ResponseDto<object> { Message = "Basket deleted successfully" };
+        }
+
+        public async Task<ResponseDto<object>> CreateOrderAsync(string userId, OrderDetailsDto orderDetailsDto, CancellationToken cancellationToken = default)
+        {
+            var basket = await _basketRepository.GetBasketAsync(userId, cancellationToken);
+
+            if (basket is null)
+            {
+                throw new BasketNotFoundException(userId);
+            }
+
+            var orderCreateDto = new OrderCreateDto
+            {
+                UserId = userId,
+                ProductIds = basket.Items.Select(basketItem => basketItem.ProductId).ToArray(),
+                Total = basket.TotalPrice,
+                Comment = orderDetailsDto.Comment,
+                DeliveryAddress = orderDetailsDto.DeliveryAddress,
+                PhoneNumber = orderDetailsDto.PhoneNumber
+            };
+
+
+            //TODO: Replace this temporary synchronous communication with RabbitMQ message bus
+            var apiUrl = "https://localhost:7004/api/orders";
+
```

Comment:
create separate method

Reply:
later on i will get rid of this code and publish message to rabbitmw instead